pkg/router/template/template_helper.go

```diff
@@ -31,9 +32,45 @@ func isTrue(s string) bool {
 	return v
 }
 
+// compiledRegexp is the store of already compiled regular
+// expressions.
+var compiledRegexp sync.Map
+
+// cachedRegexpCompile will compile pattern as a regexp and add it to
+// the compiledRegexp store if it is not already present. It will
+// return an error error if pattern is an invalid regular expression.
+// If pattern already exists in the store then no compilation is
+// necessary and the existing compiled regexp is returned. This
+// provides a huge performance improvement as repeated calls to
+// compiling a regular expression is enormous.
+// See: https://bugzilla.redhat.com/show_bug.cgi?id=1937972
+func cachedRegexpCompile(pattern string) (*regexp.Regexp, error) {
+	v, ok := compiledRegexp.Load(pattern)
+	if !ok {
+		log.V(7).Info("compiling regexp", "pattern", pattern)
+		re, err := regexp.Compile(pattern)
+		if err != nil {
+			return nil, err
+		}
+		compiledRegexp.Store(pattern, re)
+		return re, nil
+	}
+	return v.(*regexp.Regexp), nil
+}
+
+func matchString(pattern string, s string) (bool, error) {
+	status, err := regexp.MatchString(`\A(?:`+pattern+`)\z`, s)
```

Comment:
What's this line for?  Copy and paste error?

Reply:
Errr. yes. Removing. :)